src/main/java/io/apicurio/datamodels/deref/AsyncApi2NodeImporter.java

```diff
@@ -0,0 +1,227 @@
+            inlineComponent(componentType, node);
+        } else {
+            AsyncApiComponents components = ensureAsyncApiComponents();
+            String name = generateNodeName(getNameHintFromRef("ImportedSchema"), getComponentNames(components.getSchemas()));
+            components.addSchema(name, node);
+            node.attach(components);
+            setPathToImportedNode(node, componentType, name);
+        }
+    }
+
+    @Override
+    public void visitSecurityScheme(SecurityScheme node) {
+        String componentType = "securitySchemes";
+        if (shouldInline()) {
+            inlineComponent(componentType, node);
+        } else {
+            AsyncApiComponents components = ensureAsyncApiComponents();
+            String name = generateNodeName(getNameHintFromRef("ImportedSecurityScheme"), getComponentNames(components.getSecuritySchemes()));
+            components.addSecurityScheme(name, node);
+            node.attach(components);
+            setPathToImportedNode(node, componentType, name);
+        }
+    }
+
+    @Override
+    public void visitServer(Server node) {
+        // Note: there is no place in #/components to store servers, so they must be inlined.
+        ObjectNode json = Library.writeNode(node);
+        Library.readNode(json, getNodeWithUnresolvedRef());
+        setPathToImportedNode(getNodeWithUnresolvedRef(), null);
+    }
+
+    @Override
+    public void visitServerBindings(AsyncApiServerBindings node) {
+        String componentType = "serverBindings";
+        if (shouldInline()) {
+            inlineComponent(componentType, node);
+        } else {
+            AsyncApiComponents components = ensureAsyncApiComponents();
+            String name = generateNodeName(getNameHintFromRef("ImportedServerBinding"), getComponentNames(components.getServerBindings()));
+            components.addServerBinding(name, node);
+            node.attach(components);
+            setPathToImportedNode(node, componentType, name);
+        }
+    }
+
+    @Override
+    public void visitServerVariable(ServerVariable node) {
+        // Note: there is no place in #/components to store server variables, so they must be inlined.
+        ObjectNode json = Library.writeNode(node);
+        Library.readNode(json, getNodeWithUnresolvedRef());
+        setPathToImportedNode(getNodeWithUnresolvedRef(), null);
+    }
+
+    private AsyncApiComponents ensureAsyncApiComponents() {
+        AsyncApiDocument doc = (AsyncApiDocument) getDoc();
+        if (doc.getComponents() == null) {
+            doc.setComponents(doc.createComponents());
+        }
+        return doc.getComponents();
+    }
+
+    private void inlineComponent(String componentType, Node node) {
+        ObjectNode json = Library.writeNode(node);
+        Library.readNode(json, getNodeWithUnresolvedRef());
+        setPathToImportedNode(getNodeWithUnresolvedRef(), componentType, getComponentName(getNodeWithUnresolvedRef()));
+    }
```

Comment:
This method is basically the same here and in the other importer classes. Might be worth it to push to the parent class.

Reply:
Good call.  It shall be done.